unit_testing/cli/test_cli_sct_extract_metric.py

```diff
@@ -1,14 +1,23 @@
-from pytest_console_scripts import script_runner
+import csv
+import numpy as np
+
 import pytest
 import logging
 
+from spinalcordtoolbox.scripts import sct_extract_metric
+
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.script_launch_mode('subprocess')
-def test_sct_extract_metric_backwards_compat(script_runner):
-    ret = script_runner.run('sct_testing', '--function', 'sct_extract_metric')
-    logger.debug(f"{ret.stdout}")
-    logger.debug(f"{ret.stderr}")
-    assert ret.success
-    assert ret.stderr == ''
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_extract_metric_weighted_average_value_against_groundtruth():
+    """Verify that computed WA value is equivalent to known ground truth value."""
+    fname_out = 'quantif_mtr.csv'
+    sct_extract_metric.main(argv=['-i', 'mt/mtr.nii.gz', '-f', 'mt/label/atlas', '-method', 'wa', '-l', '51',
+                                  '-z', '1:2', '-o', fname_out])
+    with open(fname_out, 'r') as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=',')
+        mtr_result = np.float([row['WA()'] for row in reader][0])
```

Comment:
In this case, `np.genfromtext` is a little awkward, because the csv file contains both numeric and non-numeric columns. We want to access specifically the `WA()` column, but if we try using `np.genfromtxt`, we get this:

![Screenshot from 2021-06-21 11-14-48](https://user-images.githubusercontent.com/16181459/122786077-ea647b80-d281-11eb-8f54-26fc72cddbea.png)

We could access the value using a numeric index, but that could be error prone if we add a new column... so, I think I like using `csv.DictReader` here.

Reply:
Actually, now that I type that out, it's a bit silly that we're only checking the weighted-average column. We could easily be checking all three.

![Screenshot from 2021-06-21 11-17-30](https://user-images.githubusercontent.com/16181459/122786419-4af3b880-d282-11eb-98e4-5acf70a773db.png)

Reply:
d06dfc2

Reply:
```
     results = results[~np.isnan(results)]  # Remove non-numeric fields such as filename, SCT version, etc.
```

Nicely played :clap: